roles/ceph-osd/tasks/check_devices.yml

```diff
@@ -4,8 +4,9 @@
 # it should exist we rc=0 and don't do anything unless we do something like --force
 # As as a final word, I prefer to keep the partition check instead of running ceph-disk prepare with "failed_when: false"
 # I believe it's safer
+
 - name: check if the device is a partition or a disk
-  shell: "echo '{{ item }}' | egrep '/dev/(([a-z]{3,4}[0-9]$)|(cciss/c[0-9]{1}d[0-9]{1}p[0-9]$))'"
+  shell: "echo '{{ item }}' | egrep '/dev/[sd[a-z]{1,2}|hd[a-z]{1,2}|cciss/c[0-9]d[0-9]p|nvme[0-9]n[0-9]p][0-9]{1,2}'"
```

Comment:
Has anyone seen this?  When I run

ansible -m setup hp60ds3 says that device /dev/nvme0n1 (Intel P3700 SSD) has no partitions, but it has 58 of them.  Is this a bug in ansible module or task, or is this a problem with ceph-ansible?  I tend to think it's not anything to do with ceph-ansible.  I'm using RHEL7.2 RC4 and ansible-1.9.2-1.el7.noarch.  Also the model and vendor look wrong. 

-ben

hp60ds3 | success >> {
    "ansible_facts": {
...
        "ansible_devices": {
            "nvme0n1": {
...
                "model": null,
                "partitions": {},
                "removable": "0",
                "rotational": "0",
                "scheduler_mode": "",
                "sectors": "1562824368",
                "sectorsize": "512",
                "size": "745.21 GB",
                "support_discard": "512",
                "vendor": null
            },
...